File: system-test/java/src/main/java/com/optivem/eshop/dsl/core/usecase/shop/usecases/base/BaseShopUseCase.java
package com.optivem.eshop.dsl.core.usecase.shop.usecases.base;

import com.optivem.eshop.dsl.driver.port.shop.ShopDriver;
import com.optivem.eshop.dsl.core.shared.BaseUseCase;
import com.optivem.eshop.dsl.core.shared.UseCaseContext;

public abstract class BaseShopUseCase<TResponse, TVerification> extends BaseUseCase<ShopDriver, TResponse, TVerification> {
    protected BaseShopUseCase(ShopDriver driver, UseCaseContext context) {
        super(driver, context);
    }
}

File: system-test/typescript/src/testkit/dsl/core/usecase/shop/usecases/base/BaseShopUseCase.ts
import type { ShopDriver } from '../../../../../../driver/port/shop/shop-driver.js';
import { BaseUseCase } from '../../../../shared/base-use-case.js';

export abstract class BaseShopUseCase<TResponse, TVerification> extends BaseUseCase<
  ShopDriver,
  TResponse,
  TVerification
> {}
